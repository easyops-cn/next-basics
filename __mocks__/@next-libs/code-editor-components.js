import React from "react";

export function CodeEditorItem() {
  return <div>code editor</div>;
}
