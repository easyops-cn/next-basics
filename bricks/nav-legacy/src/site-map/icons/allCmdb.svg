<?xml version="1.0" encoding="UTF-8"?>
<svg width="28px" height="28px" viewBox="0 0 28 28" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 61 (89581) - https://sketch.com -->
    <title>编组 19备份</title>
    <desc>Created with Sketch.</desc>
    <g id="最终版" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="编组-19备份">
            <rect id="矩形" stroke="#979797" x="0.5" y="0.5" width="27" height="27" rx="6"></rect>
            <g id="编组-5" transform="translate(6.000000, 6.000000)" fill="#979797" fill-rule="nonzero">
                <g id="编组-7" transform="translate(0.000000, -0.000000)">
                    <g id="编组">
                        <path d="M8.84496615,13.5285808 L2.45106737,13.5285808 C1.67132234,13.5277766 1.03975323,12.8952293 1.04015578,12.115484 L1.04015578,2.45252417 C1.04015578,1.67313573 1.67095053,1.04161258 2.45033897,1.04015578 L12.115484,1.04015578 C12.4906049,1.0392175 12.8504936,1.18847161 13.1148493,1.45461561 C13.3814439,1.72121016 13.527124,2.07521276 13.527124,2.45325257 L13.505272,8.84496615 L13.505272,8.84642295 C13.5050117,9.03248912 13.6040364,9.20456018 13.7650443,9.29781863 C13.9260522,9.39107708 14.1245824,9.39135474 14.2858506,9.29854703 C14.4471187,9.20573931 14.5466243,9.03394592 14.5468845,8.84787975 L14.5680082,2.45470937 L14.5680082,2.45252417 C14.5682014,1.80214131 14.3099732,1.17833032 13.8501514,0.718371864 C13.3903295,0.258413413 12.7665953,0 12.1162124,0 L2.45033897,0 C1.80021618,0 1.176742,0.258359824 0.717172406,0.718202648 C0.257602811,1.17804547 -0.000386518083,1.8016731 -9.76950676e-07,2.45179577 L-9.76950676e-07,12.1162124 C-0.000580486809,12.7666468 0.257546555,13.390608 0.717473348,13.8505348 C1.17740014,14.3104616 1.80136132,14.5685886 2.45179577,14.5680091 L8.84496615,14.5680091 C9.13199609,14.5680091 9.36467984,14.3353244 9.36467984,14.0482945 C9.36467984,13.7612645 9.13199609,13.5285808 8.84496615,13.5285808 Z" id="路径"></path>
                        <path d="M11.45701,4.67487382 C11.7454479,4.67487382 11.9792731,4.44104867 11.9792731,4.15261072 C11.9792731,3.86417278 11.7454479,3.63034763 11.45701,3.63034763 L3.15688736,3.62961923 C3.01863444,3.62981205 2.88613136,3.6849678 2.78857684,3.78293221 C2.69102232,3.88089662 2.63642263,4.0136298 2.63680744,4.15188232 C2.63680744,4.44032887 2.86989761,4.6741454 3.15688736,4.67414542 L11.45701,4.67633062 L11.45701,4.67487382 Z" id="路径"></path>
                        <path d="M6.2693423,7.79679796 C6.55560367,7.79679796 6.78723705,7.56370983 6.78723705,7.27672007 C6.78762185,7.1390471 6.73325239,7.00686973 6.63610767,6.90931512 C6.53896296,6.81176052 6.40701568,6.756835 6.2693423,6.75664218 L3.15688736,6.75664218 C2.96715373,6.75001573 2.78904225,6.84787918 2.69289631,7.01158232 C2.59633097,7.17515212 2.59633097,7.37828803 2.69289631,7.54185782 C2.78904226,7.70556096 2.96715373,7.80342441 3.15688736,7.79679796 L6.2693423,7.79752636 L6.2693423,7.79679796 Z" id="路径"></path>
                        <path d="M3.15907257,9.88075152 C3.02133248,9.88094457 2.88931056,9.93584682 2.79205021,10.0333804 C2.69478986,10.1309139 2.64025815,10.2630893 2.64045096,10.4008294 C2.64045096,10.6878192 2.8720828,10.9209073 3.15834417,10.9209073 L5.23282853,10.9209073 C5.42241058,10.9276827 5.60048595,10.8301186 5.69681958,10.6666956 C5.79351499,10.5029004 5.79351499,10.2994868 5.69681958,10.1356917 C5.60048595,9.97226862 5.42241058,9.87470457 5.23282853,9.88147992 L3.15834417,9.88147992 L3.15907257,9.88075152 Z" id="路径"></path>
                    </g>
                    <g id="编组" transform="translate(8.784689, 9.516746)">
                        <path d="M2.87593985,2.39389311 C2.87593985,2.82707613 3.22710407,3.17824035 3.66028708,3.17824035 C4.0934701,3.17824035 4.44463431,2.82707613 4.44463431,2.39389311 C4.44463431,1.9607101 4.0934701,1.60954588 3.66028708,1.60954588 C3.22710407,1.60954588 2.87593985,1.9607101 2.87593985,2.39389311 Z" id="路径"></path>
                        <path d="M3.66028708,3.98873248 C2.50947579,3.98858179 1.44370422,3.38269115 0.854938483,2.39389311 C1.44378003,1.40517243 2.50950348,0.799309143 3.66028708,0.799053742 C4.81107068,0.799309143 5.87679413,1.40517243 6.46563568,2.39389311 C5.87686994,3.38269115 4.81109838,3.98858179 3.66028708,3.98873248 M6.8280041,1.5190845 C6.05707576,0.558716349 4.89180519,0.000317461375 3.66028708,0.00110950406 C2.13469001,-0.000262894417 0.737282419,0.854303263 0.0434005468,2.21297035 C-0.0144396795,2.3266513 -0.0144396795,2.46113493 0.0434005468,2.57481587 C0.737282419,3.93348296 2.13469001,4.78804912 3.66028708,4.78667672 C5.18578257,4.78768776 6.58301636,3.93322789 7.27717362,2.57481587 C7.33502121,2.46095238 7.33502121,2.32631094 7.27717362,2.21244745 C7.15150239,1.96660492 7.00099471,1.73427293 6.8280041,1.5190845" id="形状"></path>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>