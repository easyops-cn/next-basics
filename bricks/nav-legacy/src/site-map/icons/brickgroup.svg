<svg viewBox="0 0 11 12" xmlns="http://www.w3.org/2000/svg"><path d="M10.5 10.25a1.75 1.75 0 1 1-2.242-1.68V6.827L5.246 5.816l-3.01 1.01v1.742a1.75 1.75 0 1 1-.985.004V6.427l3.507-1.515V3.43A1.751 1.751 0 0 1 5.25 0a1.75 1.75 0 0 1 .492 3.43v1.49l3.504 1.503-.004 2.147a1.751 1.751 0 0 1 1.258 1.68zm-9.516 0a.766.766 0 1 0 1.533-.001.766.766 0 0 0-1.533.001zm7.766-.766a.766.766 0 1 0 .001 1.533.766.766 0 0 0-.001-1.533z" fill="#595959"/></svg>