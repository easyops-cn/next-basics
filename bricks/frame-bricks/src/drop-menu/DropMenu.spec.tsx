import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import {
  DropMenu,
  filterBySearch,
  matchSearchValue,
  SubCategory,
} from "./DropMenu";
import { mount, shallow } from "enzyme";
import { DownOutlined, UpOutlined, TagOutlined } from "@ant-design/icons";
import { GeneralIcon, Link } from "@next-libs/basic-components";
import { Button, Input, Popover } from "antd";

describe("DropMenu", () => {
  const menuData = [
    {
      title: "xxx",
      children: [{ name: "xxx", to: "/ddd" }],
      subCategory: [
        {
          title: "xxx",
          children: [
            {
              name: "xxx",
              to: "/ddd",
            },
          ],
        },
      ],
    },
  ];
  const menuDataTwo = [
    {
      title: "xxx",
      children: [{ name: "xxx", to: "/ddd" }],
      subCategory: [
        {
          title: "xxx",
          children: [
            {
              name: "xxx",
              to: "/ddd",
            },
          ],
        },
      ],
    },
    {
      title: "xxx",
      children: [{ name: "aaa", to: "/ddd" }],
      subCategory: [
        {
          title: "xxx",
          children: [
            {
              name: "aaa",
              to: "/ddd",
            },
          ],
        },
      ],
    },
  ];
  it("should work", () => {
    const wrapper = mount(<DropMenu menuData={menuData} title="xxx" />);
    const handleToggle = jest.fn();
    const div = wrapper.find("#menuPopover");
    expect(wrapper.find(DownOutlined)).toHaveLength(0);
    expect(wrapper.find(UpOutlined)).toHaveLength(0);
    expect(wrapper.find(Popover)).toHaveLength(1);
    expect(wrapper.find(Link)).toHaveLength(0);
    expect(wrapper.find("#menuPopover")).toHaveLength(1);
    expect(wrapper.find(Popover).invoke("getPopupContainer")(null)).toEqual(
      null
    );
  });

  it("should work with SubCategory", () => {
    const subCategory = [
      {
        title: "xxx",
        children: [
          {
            name: "aaa",
            to: "/ddd",
          },
        ],
      },
    ];
    const wrapper = mount(<SubCategory {...subCategory} />);
    // expect(wrapper).toMatchSnapshot();
    const toggleBtn = wrapper.find(Button);
    expect(wrapper.find(DownOutlined)).toHaveLength(0);
    expect(wrapper.find(UpOutlined)).toHaveLength(1);
    expect(wrapper.find(Link)).toHaveLength(0);
    toggleBtn.simulate("click");
    expect(wrapper.find(DownOutlined)).toHaveLength(1);
    expect(wrapper.find(UpOutlined)).toHaveLength(0);
  });

  it("should toMatchSnapshot work", () => {
    const wrapper = mount(<DropMenu menuData={menuData} title="xxx" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should work with matchSearchValue", () => {
    expect(matchSearchValue({ to: "/APP", name: "app" }, "app")).toBe(true);
    expect(matchSearchValue({ to: "/APP", name: "应用" }, "应用2")).toBe(false);
  });

  it("should work with filterBySearch", () => {
    expect(filterBySearch(menuDataTwo, "xxx")).toEqual([
      {
        title: "xxx",
        children: [{ name: "xxx", to: "/ddd" }],
        subCategory: [
          {
            title: "xxx",
            children: [
              {
                name: "xxx",
                to: "/ddd",
              },
            ],
          },
        ],
      },
    ]);
    expect(filterBySearch(menuData, "")).toEqual(menuData);
  });
});
