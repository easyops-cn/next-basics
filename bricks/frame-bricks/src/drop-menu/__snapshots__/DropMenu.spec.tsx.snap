// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropMenu should toMatchSnapshot work 1`] = `
<DropMenu
  menuData={
    Array [
      Object {
        "children": Array [
          Object {
            "name": "xxx",
            "to": "/ddd",
          },
        ],
        "subCategory": Array [
          Object {
            "children": Array [
              Object {
                "name": "xxx",
                "to": "/ddd",
              },
            ],
            "title": "xxx",
          },
        ],
        "title": "xxx",
      },
    ]
  }
  title="xxx"
>
  <div
    className="menuPopover"
    id="menuPopover"
  >
    <Popover
      content={[Function]}
      getPopupContainer={[Function]}
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      overlayClassName="popoverInMenu"
      overlayStyle={Object {}}
      placement="bottom"
      title={null}
      transitionName="zoom-big"
      trigger="hover"
    >
      <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        getPopupContainer={[Function]}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        overlay={
          <React.Fragment>
            <div
              className="ant-popover-inner-content"
            >
              <div
                className="menuMainContainer"
              >
                <div
                  className="menuSearchContainer"
                >
                  <Input
                    onChange={[Function]}
                    placeholder="输入关键词搜索"
                    prefix={
                      <ForwardRef(SearchOutlined)
                        className="searchIcon"
                      />
                    }
                    type="text"
                  />
                </div>
                <div
                  className="masonry"
                >
                  <div
                    className="column"
                  >
                    <div
                      className="newItem"
                    >
                      <div
                        className="subCategoryContainer"
                      >
                        <ForwardRef(TagOutlined)
                          className="categoryIcon"
                          style={
                            Object {
                              "marginRight": "4px",
                            }
                          }
                        />
                        <span
                          className="categoryTitle"
                          style={
                            Object {
                              "fontSize": undefined,
                            }
                          }
                        >
                          xxx
                        </span>
                      </div>
                      <SubCategory
                        title="xxx"
                      >
                        Object {
                          "name": "xxx",
                          "to": "/ddd",
                        }
                      </SubCategory>
                      <div
                        style={
                          Object {
                            "height": "5pxß",
                          }
                        }
                      />
                      <div
                        className="objectItemContainer categoryNameContainer"
                      >
                        <Link
                          href="/ddd"
                          style={
                            Object {
                              "fontSize": undefined,
                            }
                          }
                        >
                          xxx
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </React.Fragment>
        }
        overlayClassName="popoverInMenu"
        overlayStyle={Object {}}
        placement="bottom"
        prefixCls="ant-popover"
        transitionName="zoom-big"
        trigger="hover"
      >
        <ForwardRef(Tooltip)
          arrowContent={
            <span
              className="ant-popover-arrow-content"
            />
          }
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          builtinPlacements={
            Object {
              "bottom": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tc",
                  "bc",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "bottomLeft": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "bl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "bottomRight": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tr",
                  "br",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "left": Object {
                "ignoreShake": true,
                "offset": Array [
                  -4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "cr",
                  "cl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "leftBottom": Object {
                "ignoreShake": true,
                "offset": Array [
                  -4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "br",
                  "bl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "leftTop": Object {
                "ignoreShake": true,
                "offset": Array [
                  -4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tr",
                  "tl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "right": Object {
                "ignoreShake": true,
                "offset": Array [
                  4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "cl",
                  "cr",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "rightBottom": Object {
                "ignoreShake": true,
                "offset": Array [
                  4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "bl",
                  "br",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "rightTop": Object {
                "ignoreShake": true,
                "offset": Array [
                  4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "tr",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "top": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  -4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "bc",
                  "tc",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "topLeft": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  -4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "bl",
                  "tl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "topRight": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  -4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "br",
                  "tr",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
            }
          }
          getPopupContainer={[Function]}
          getTooltipContainer={[Function]}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          onPopupAlign={[Function]}
          onVisibleChange={[Function]}
          overlay={
            <React.Fragment>
              <div
                className="ant-popover-inner-content"
              >
                <div
                  className="menuMainContainer"
                >
                  <div
                    className="menuSearchContainer"
                  >
                    <Input
                      onChange={[Function]}
                      placeholder="输入关键词搜索"
                      prefix={
                        <ForwardRef(SearchOutlined)
                          className="searchIcon"
                        />
                      }
                      type="text"
                    />
                  </div>
                  <div
                    className="masonry"
                  >
                    <div
                      className="column"
                    >
                      <div
                        className="newItem"
                      >
                        <div
                          className="subCategoryContainer"
                        >
                          <ForwardRef(TagOutlined)
                            className="categoryIcon"
                            style={
                              Object {
                                "marginRight": "4px",
                              }
                            }
                          />
                          <span
                            className="categoryTitle"
                            style={
                              Object {
                                "fontSize": undefined,
                              }
                            }
                          >
                            xxx
                          </span>
                        </div>
                        <SubCategory
                          title="xxx"
                        >
                          Object {
                            "name": "xxx",
                            "to": "/ddd",
                          }
                        </SubCategory>
                        <div
                          style={
                            Object {
                              "height": "5pxß",
                            }
                          }
                        />
                        <div
                          className="objectItemContainer categoryNameContainer"
                        >
                          <Link
                            href="/ddd"
                            style={
                              Object {
                                "fontSize": undefined,
                              }
                            }
                          >
                            xxx
                          </Link>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </React.Fragment>
          }
          overlayClassName="popoverInMenu"
          overlayStyle={Object {}}
          placement="bottom"
          prefixCls="ant-popover"
          transitionName="zoom-big"
          trigger="hover"
          visible={false}
        >
          <Trigger
            action="hover"
            afterPopupVisibleChange={[Function]}
            arrowContent={
              <span
                className="ant-popover-arrow-content"
              />
            }
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            autoDestroy={false}
            blurDelay={0.15}
            builtinPlacements={
              Object {
                "bottom": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tc",
                    "bc",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "bottomLeft": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tl",
                    "bl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "bottomRight": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tr",
                    "br",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "left": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    -4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "cr",
                    "cl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "leftBottom": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    -4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "br",
                    "bl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "leftTop": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    -4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tr",
                    "tl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "right": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "cl",
                    "cr",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "rightBottom": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "bl",
                    "br",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "rightTop": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tl",
                    "tr",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "top": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    -4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "bc",
                    "tc",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "topLeft": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    -4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "bl",
                    "tl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "topRight": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    -4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "br",
                    "tr",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
              }
            }
            defaultPopupVisible={false}
            destroyPopupOnHide={false}
            focusDelay={0}
            getDocument={[Function]}
            getPopupClassNameFromAlign={[Function]}
            getPopupContainer={[Function]}
            hideAction={Array []}
            mask={false}
            maskClosable={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            onPopupAlign={[Function]}
            onPopupVisibleChange={[Function]}
            overlay={
              <React.Fragment>
                <div
                  className="ant-popover-inner-content"
                >
                  <div
                    className="menuMainContainer"
                  >
                    <div
                      className="menuSearchContainer"
                    >
                      <Input
                        onChange={[Function]}
                        placeholder="输入关键词搜索"
                        prefix={
                          <ForwardRef(SearchOutlined)
                            className="searchIcon"
                          />
                        }
                        type="text"
                      />
                    </div>
                    <div
                      className="masonry"
                    >
                      <div
                        className="column"
                      >
                        <div
                          className="newItem"
                        >
                          <div
                            className="subCategoryContainer"
                          >
                            <ForwardRef(TagOutlined)
                              className="categoryIcon"
                              style={
                                Object {
                                  "marginRight": "4px",
                                }
                              }
                            />
                            <span
                              className="categoryTitle"
                              style={
                                Object {
                                  "fontSize": undefined,
                                }
                              }
                            >
                              xxx
                            </span>
                          </div>
                          <SubCategory
                            title="xxx"
                          >
                            Object {
                              "name": "xxx",
                              "to": "/ddd",
                            }
                          </SubCategory>
                          <div
                            style={
                              Object {
                                "height": "5pxß",
                              }
                            }
                          />
                          <div
                            className="objectItemContainer categoryNameContainer"
                          >
                            <Link
                              href="/ddd"
                              style={
                                Object {
                                  "fontSize": undefined,
                                }
                              }
                            >
                              xxx
                            </Link>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </React.Fragment>
            }
            popup={[Function]}
            popupAlign={Object {}}
            popupClassName="popoverInMenu"
            popupPlacement="bottom"
            popupStyle={Object {}}
            popupTransitionName="zoom-big"
            popupVisible={false}
            prefixCls="ant-popover"
            showAction={Array []}
            visible={false}
          >
            <div
              className="newSiteMap"
              key="trigger"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="appName"
              >
                xxx
              </span>
            </div>
          </Trigger>
        </ForwardRef(Tooltip)>
      </Tooltip>
    </Popover>
  </div>
</DropMenu>
`;
